include/random/rand32.h

```diff
@@ -8,28 +8,81 @@
  * @file
  * @brief Random number generator header file
  *
- * This header file declares prototypes for the kernel's random number generator
- * APIs.
+ * This header file declares prototypes for the kernel's random number
+ * generator APIs.
  *
- * Typically, a platform enables the hidden CUSTOM_RANDOM_GENERATOR or
+ * Typically, a platform enables the appropriate source for the random
+ * number generation based on the hardware platform's capabilities or
  * (for testing purposes only) enables the TEST_RANDOM_GENERATOR
- * configuration option and provide its own driver that implements
- * sys_rand32_get().
+ * configuration option.
  */
 
 #ifndef ZEPHYR_INCLUDE_RANDOM_RAND32_H_
 #define ZEPHYR_INCLUDE_RANDOM_RAND32_H_
 
 #include <zephyr/types.h>
 
+/**
+ * @brief Random Function APIs
+ * @defgroup random_api Random Function APIs
+ * @{
+ */
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @brief Return a 32-bit random value that should pass general
+ * randomness tests.
+ *
+ * @note The random value returned is not a cryptographically secure
+ * random number value.
+ *
+ * @return 32-bit random value.
+ */
 extern u32_t sys_rand32_get(void);
+/**
+ * @brief Fill the destination buffer with random data values that should
+ * pass general randomness tests.
+ *
+ * @note The random values returned are not considered cryptographically
+ * secure random number values.
+ *
+ * @param [out] dst destination buffer to fill with random data.
+ * @param len size of the destination buffer.
+ *
+ */
+extern void sys_rand_get(void *dst, u32_t len);
+
+/**
+ * @brief Generate and return a 32-bit cryptographically secure random value.
+ *
+ * @note If the random value requested does not need to be cryptographically
+ * secure then use sys_rand32_get() instead.
+ *
+ * @return 32-bit random value.
+ */
+extern u32_t sys_csrand32_get(void);
+/**
+ * @brief Fill the destination buffer with cryptographically secure
+ * random data values.
+ *
+ * @note If the random values requested do not need to be cryptographically
+ * secure then use sys_rand_get() instead.
+ *
+ * @param [out] dst destination buffer to fill.
+ * @param len size of the destination buffer.
+ *
+ */
+extern void sys_csrand_get(void *dst, u32_t len);
```

Comment:
This function needs to return status. It can fail, and clients must know that has happened. Otherwise failure can be used to give less random data than requested. For example, if a client requests more data than the DRBG allows, this call will fail, and the client buffer will be completely unwritten.

You could argue that this call could panic if called inappropriately, but ignoring errors isn't possible from a security perspective.